src/categories/data.js

```diff
@@ -7,13 +7,15 @@ var winston = require('winston');
 var db = require('../database');
 
 module.exports = function (Categories) {
-	Categories.getCategoryData = function (cid, callback) {
+	Categories.getCategoryData = function (cid, callback, isModificationBlocked) {
 		async.waterfall([
 			function (next) {
 				db.getObject('category:' + cid, next);
 			},
 			function (category, next) {
-				modifyCategory(category);
+				if (!isModificationBlocked) {
```

Comment:
Let's please document the reason, why do we need to place this check.

Reply:
Comment added for documentation purposes.

Reply:
@arbisofttaha Actually I wanted to document the exact edge case we tackled here.